pkg/client/request.go

```diff
@@ -79,6 +79,14 @@ func (r *RequestConstructionError) Error() string {
 	return fmt.Sprintf("request construction error: '%v'", r.Err)
 }
 
+type PatchType string
```

Comment:
I don't think so... pkg/apiserver/patch is only one implementation. The header supports all three implementations.

The bigger question is whether it goes in the server or the client. For a proper API it probably needs to be duplicated in both.